pkg/apis/kubeone/types.go

```diff
@@ -111,6 +111,30 @@ type ClusterNetworkConfig struct {
 	ServiceSubnet     string `json:"serviceSubnet"`
 	ServiceDomainName string `json:"serviceDomainName"`
 	NodePortRange     string `json:"nodePortRange"`
+	CNI               *CNI   `json:"cni,omitempty"`
+}
+
+// CNIProvider type
+type CNIProvider string
+
+// List of CNI Providers
+const (
+	// https://docs.projectcalico.org/v3.7/getting-started/kubernetes/installation/flannel
+	// This provider does not support encryption.
+	CNIProviderCanal = CNIProvider("canal")
+
+	// https://www.weave.works/docs/net/latest/kubernetes/kube-addon/
+	// This provider support optional encryption.
+	// Strong secret will be autogenerated.
+	CNIProviderWeaveNet = CNIProvider("weave-net")
+)
+
+// CNI config
+type CNI struct {
+	// Provider choise
+	Provider CNIProvider `json:"provider"`
+	// Some of providers do provide optional encryption
```

Comment:
A valid godoc comment is preferred. `Encrypted enables encryption for supported CNI plugins` or something similar could work.